src/sssom_schema/schema/sssom_schema.yaml

```diff
@@ -118,15 +119,17 @@ slots:
   subject_id:
     description: The ID of the subject of the mapping.
     range: EntityReference
-    required: true
     mappings:
     - owl:annotatedSource
     slot_uri: owl:annotatedSource
     examples:
       - value: HP:0009894
         description: The CURIE denoting the Human Phenotype Ontology concept of 'Thickened ears'
-  literal:
+  subject_literal:
```

Comment:
I was rather envisioning reusing `subject_label` rather than creating a new slot.

Reply:
I know, but subject_label has a very different purpose than subject_literal. Your proposal effectively _changes_ the purpose of subject_label if the "literal mode" is switched on, so I can probably be convinced to using subject_label as you say. My intention was to seperate the "literal" and "entity" modes a tiny bit more cleanly, but probably, I am just introducing churn... One more push with a ring finger and you will have me convinced!

Reply:
> Your proposal effectively changes the purpose of subject_label if the "literal mode" is switched on

Yes it does. And I think it is a better solution (not perfect, but better) than adding a new slot that is only meaningful when the “literal mode is on” (i.e. when `subject_type == sssom literal`).

If we reuse `subject_label`, all we have to do is to say that `subject_label` is normally the label of the entity being mapped, or the literal being mapped if the mapping is a literal mapping. Full stop.

If we create `subject_literal`, we not only need to explain what it is (the literal being mapped), but also deal with the possible incorrect use of both `subject_literal` (which must not be used for non-literal mappings) and `subject_label` (which must not be used for literal mappings). Another needless complication.

Reply:
> My intention was to seperate the "literal" and "entity" modes a tiny bit more cleanly

I am going to be blunt (blunter than usual, I mean :P ): the SSSOM data model has _not_ been designed to separate things cleanly.

A “clean design” would not have put all the metadata slots directly in the `mapping` class, resulting in all those slots that exist in two versions (one for the subject side, another for the object side) such as `subject_label` and `object_label`, `subject_preprocessing` and `object_preprocessing`, etc.

Instead, a clean design would have another class specifically to represent an entity being mapped (let’s call it `entity`), _without consideration for whether it is the subject or the object`. That class would have had slots such as `id`, `label`, `preprocessing`, `source`, etc. And the `mapping` class would have had two slots `subject` and `object`, both accepting a value of type `entity`.

This would have been much cleaner, would have make manipulating the mappings and their entities much easier (inverting a mapping, for example, would simply have involved swapping the `subject` and the `object` slot, instead of having to swap all the `subject_*` and `object_*` slots). It would also have made extending the model to deal with new types of entities easier as well, since it would just have required adding new subclasses of `entity`. We could even have dispensed with the `subject|object_type` slots, since the type of an entity would have been naturally indicated by which subclass of `entity` is used.

My understanding, based on some old discussions that I have read in this repo, is that such a design has been deliberately _avoided_ because there was a clear preference for a data model that was much closer to the TSV serialisation (i.e., where each column of the TSV maps directly to one slot in the `mapping` class).

I do not criticize that choice, but now we have to stick to the “flat” model that we have as a result. Trying to introduce some clean object-oriented design on top of it would only make things more confusing. (Hence my refusal to the entire idea of having a separate `literal mapping` class.)